packages/mongodb-rag-ingest/src/embed/updateEmbeddedContent.test.ts

```diff
@@ -138,4 +146,64 @@ describe("updateEmbeddedContent", () => {
       "2cbfe9901657ca15260fe7f58c3132ac1ebd0d610896082ca1aaad0335f2e3f1"
     );
   });
-});
+  it("processes pages concurrently", async () => {
+    const pageStore = makeMockPageStore();
+    const concurrentPages: Page[] = [
+      { ...examplePage, url: 'https://example.com/test1'},
+      { ...examplePage, url: 'https://example.com/test2'},
+      { ...examplePage, url: 'https://example.com/test3'}
+    ];
+
+    await persistPages({
+      pages: concurrentPages,
+      store: pageStore,
+      sourceName: "test",
+    });
+
+    const embeddedContentStore = makeMockEmbeddedContentStore();
+
+    const since = new Date("2000-01-01");
+
+    const startTimes: number[] = [];
+    const endTimes: number[] = [];
+
+    const originalDateNow = Date.now;
+    let currentFakeTime = originalDateNow();
+
+    jest.spyOn(global.Date, 'now').mockImplementation(() => currentFakeTime);
+
+    const mockEmbedder: jest.Mocked<Embedder> = {
+      embed: jest.fn().mockImplementation(async (param) => {
+        const startTime = Date.now();
+        startTimes.push(startTime);
+        await new Promise((resolve) => setTimeout(resolve, Math.random() * 100)); // Random delay to simulate processing
+        currentFakeTime += 50; // Simulate time passage
+        const endTime = Date.now();
+        endTimes.push(endTime);
+        return { embedding: [1, 2, 3] };
+      }),
+    } as unknown as jest.Mocked<Embedder>;
+
+    await updateEmbeddedContent({
+      embedder: mockEmbedder,
+      embeddedContentStore,
+      pageStore,
+      since,
+      embedConcurrencyOptions: { processPages: 2, createChunks: 2 },
```

Comment:
i am not sure if you're testing both of these. 

anyways, i think these should be separate tests, one for `processPages` and another for `createChunks`